rwilliams659/beercation
# Rw user saved breweries component

**What changes/additions are a part of this PR?**  
The main part of this adds the UserSavedBreweries & UserSavedBreweryCard components. These are rendered on the To Visit & Visited pages, so I also added Routes in app for these. 
I was originally going to re-use my current Breweries & BreweryCard components for this, but I realized that the cards would need to be quite a bit different, and the amount of conditional rendering I would need to use in the existing Breweries component would have made it extremely messy and difficult to maintain, so I instead created new components. 
I also realized during this process that the way I was rendering a Link around each complete BreweryCard component (as we were taught) meant that the "Mark as to visit/visited" buttons on those cards also were linked, which I did not want. I had to re-do the Links to just be wrapped around what I WANTED to be linked, and re-adjust the CSS accordingly. 
  
**Any background context you want to provide?**   
I ran into a lot of bugs along the way of implementing this, so I had to fix/re-structure some other elements of components as well. This was a bigger PR than I would have wanted, but I didn't want to push up broken code!
  
**Where should the reviewer start?**  '
UserSavedBreweries component
  
**How should this be tested (test suite/manually)?**   
Manually for now. I will need to go back in and add tests for UserSavedBreweries & UserSavedBreweryCard components, as well as audit the rest of my existing tests to make sure nothing broke/add any additional tests needed based on these changes. 
  
**What are the relevant tickets?**   
resolve #36 
resolve #37 
resolve #38 
resolve #39
